arviz/plots/backends/matplotlib/posteriorplot.py

```diff
@@ -319,18 +319,21 @@ def format_axes():
             rug=False,
             show=False,
         )
-    else:
+    elif values.dtype.kind == "i":
         if bins is None:
-            if values.dtype.kind == "i":
-                xmin = values.min()
-                xmax = values.max()
-                bins = get_bins(values)
-                ax.set_xlim(xmin - 0.5, xmax + 0.5)
-            else:
-                bins = "auto"
+            xmin = values.min()
+            xmax = values.max()
+            bins = get_bins(values)
+            ax.set_xlim(xmin - 0.5, xmax + 0.5)
         kwargs.setdefault("align", "left")
         kwargs.setdefault("color", "C0")
         ax.hist(values, bins=bins, alpha=0.35, **kwargs)
+    elif values.dtype.kind == "b":
+        if bins is None:
+            bins = "auto"
+        kwargs.setdefault("color", "C0")
+        ax.bar(["False", "True"], [(~values).sum(), values.sum()], alpha=0.35, **kwargs)
+        hdi_prob = "hide"
```

Comment:
we should add an else and an error to make sure it fails informatively if anyone ever passed an array with object dtype for example

Reply:
Adding `raise TypeError("Values Must be float, Integer or boolean")` to the else.